lib/palindrome_check.rb

```diff
@@ -1,5 +1,55 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
+
+require 'pry'
+
+#with array and string
+# def palindrome_check(my_phrase)
+#   return false if my_phrase == nil
+#   return true if my_phrase.length <= 1
+#
+#   array = my_phrase.split(//)
+#
+#   temp_string = ""
+#
+#   array.each do |char|
+#     if char != " "
+#       temp_string += char
+#     end
+#   end
+#
+#   return temp_string == temp_string.reverse
+# end
+
+#with hashes
 def palindrome_check(my_phrase)
```

Comment:
If I understand your algorithm using hashes correctly, then it will return true for more cases then just palindromes. e.g. "aabbcc" is not a palindrome but since each character is repeated twice, the hash will have values that are a multiple of 2 and hence the method will return true when it should not.